go/pools/resource_pool.go

```diff
@@ -108,16 +109,10 @@ func (rp *ResourcePool) get(wait bool, timeout time.Duration) (resource Resource
 			return nil, nil
 		}
 		startTime := time.Now()
-		if timeout == 0 {
-			wrapper, ok = <-rp.resources
-		} else {
-			tmr := time.NewTimer(timeout)
-			defer tmr.Stop()
-			select {
-			case wrapper, ok = <-rp.resources:
-			case <-tmr.C:
-				return nil, TIMEOUT_ERR
-			}
+		select {
+		case wrapper, ok = <-rp.resources:
+		case <-ctx.Done():
+			return nil, TIMEOUT_ERR
```

Comment:
This is not 100% accurate: you can look at ctx.Err and see if it's context.DeadlineExceeded vs context.Canceled. Not sure it matters in this case though.